Automattic/wp-calypso
# Manage: New Sites Panel using the DataViews component from Gutenberg (Table Atelier)

Fix: https://github.com/Automattic/jetpack-manage/issues/282

## Proposed Changes

![image](https://github.com/Automattic/wp-calypso/assets/9832440/6321206c-9389-4c77-980c-74a012c7607b)

This PR is a prototype of the new DataViews (Gutenberg) implementation in Jetpack Manage as part of the first project to redesign the Sites Panel section. The main goal of this prototype is to check for potential issues and discover improvements in the DataViews components. Also, this is the starting point for the new Sites Panel redesign project (Stage 1), which is to implement the DataViews with the right side panel reusing the current components.

This implementation reuses existing components in the current Sites Dashboard. All the feature components like Stats, Backup, Plugins, etc.) and the action buttons work.

![image](https://github.com/Automattic/wp-calypso/assets/9832440/2fffd705-aadc-49dd-a7c6-9567feedf957)

I implemented a new route `/sites` behind the feature flag. Easily exchangeable with the `dashboard` route. Also, I added the new Sites Dashboard component (v2), which is an extraction of the current Sites Overview component, and removed all the unused code. 

**How to view the new design:** Go to `/sites`, which will automatically activate the feature flag for the Sites V2 menu. (Section and feature flag for the menu only available on dev, horizon, and staging).

**How to go back to the old design:** Go to `/dashboard` and reload, and it will show the current old design.

**Feature Flag:** `jetpack/manage-sites-v2-menu` To activate the Sites V2 menu. You can deactivate the feature flag using this query in the URL: `?flags=-jetpack/manage-sites-v2-menu`

## Testing Instructions

**Note:** Remember that this is a prototype behind a feature flag to set the starting point (code base) for the rest of the development. The design is close to the desired one, but it doesn't match it. It will be addressed in a future PR. We need to merge it into trunk ASAP to unblock the rest of the tasks/issues of the project. We are testing the foundation and mainly checking that this code doesn't interfere with the current code, routes, views... 

Summary:

- This PR establishes the main code that we can continue working on and a representation of how it'd look.
- The styles won't match with the Figma design, it's not the intention of this PR.
- The Preview Pane will be addressed in another PR.

**What to test?**

- Check the code as a prototype. Pay special attention to the high-level parts, folders/files, routes, and feature flags.
- Ensure that the feature flag is working properly:
   - `/dashboard` should show the old design.
   - `/sites` should show the new design with the product columns showing info.
      - Each product should work like the old version, including the action menu, except for the Get actions (`Get Score`) and the favorite star.
      - The license buttons(top-right) should work.
   - `/sites/favorites` should show only favorite sites.
   - `/sites?issue_types=all_issues` should show only sites with issues.
- Go to the `/dashboard` URL. Everything should work with the current-old design.

## Pre-merge Checklist
